### PR TITLE
Assignment 2 Huang

### PR DESCRIPTION
**Pull request by KittenLR**

### CODE REVIEW

File: week03/07_Numpy.ipynb [hunk 1]
```diff
@@ -1725,6 +1725,248 @@
     "x[i] = 99\n",
     "print(x)"
    ]
+  },
```
Comment (stargrep) [inline week03/07_Numpy.ipynb:1728]: don't change this file.

File: week03/assignment2/assignment2 - Huang.py [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+assert is_prime(7907)
+assert not is_prime(-1)
+assert not is_prime(0)
+
+
+# Q2 Write a function rotate(ar[], d) that rotates arr[] of size n by d elements.
+# Input ar = [1,2,3,4,5,6,7], d = 2
+# Output [3,4,5,6,7,1,2]
+
+def rotate(ar: [int], d: int) -> [int]:
```
Comment (stargrep) [inline week03/assignment2/assignment2 - Huang.py:29]: can do in one line

File: week03/assignment2/assignment2 - Huang.py [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+assert selection_sort([[1, 100], [2, 70], [3, 95], [4, 66], [5, 98]]) == [[4, 66], [2, 70], [3, 95], [5, 98], [1, 100]]
+
+
+# Q4. Convert a list of Tuples into Dictionary
+# tip: copy operation - copy by value, copy by reference
+
+def convert(tup: (any), di: {any, any}) -> None: 
+    for i in range(0, len(tup), 2):
+        di.setdefault(tup[i], tup[i+1])
+    pass
```
Comment (stargrep) [inline week03/assignment2/assignment2 - Huang.py:68]: I think di[k, v] would work.
remove 'pass'

**Review by stargrep — CHANGES_REQUESTED** [review]

File: week03/07_Numpy.ipynb (hunk 1 above)
Comment (KittenLR) [reply week03/07_Numpy.ipynb]: file reverted

File: week03/assignment2/assignment2 - Huang.py (hunk 2 above)
Comment (KittenLR) [reply week03/assignment2/assignment2 - Huang.py:29]: Done

**Review by stargrep — APPROVED** [review]

**Pull request merged**